### PR TITLE
Use automatic resource management to close streams

### PR DESCRIPTION
**Pull request by liketic**
We can use automatic resource management introduced in java 7, which makes code shorter.

### CODE REVIEW

File: bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/BookieStatus.java [hunk 1]
```diff
@@ -123,16 +123,9 @@ synchronized void writeToDirectories(List<File> directories) {
      * @throws IOException
      */
     private static void writeToFile(File file, String body) throws IOException {
-        FileOutputStream fos = new FileOutputStream(file);
-        BufferedWriter bw = null;
-        try {
-            bw = new BufferedWriter(new OutputStreamWriter(fos, UTF_8));
+        try (FileOutputStream fos = new FileOutputStream(file);
+             BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(fos, UTF_8))) {
```
Comment (sijie) [inline bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/BookieStatus.java]:
```suggestion
             try (BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(fos, UTF_8)))) {
```

File: bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/Cookie.java [hunk 2]
```diff
@@ -217,16 +217,9 @@ public void writeToDirectory(File directory) throws IOException {
     }
 
     public void writeToFile (File versionFile) throws IOException {
-        FileOutputStream fos = new FileOutputStream(versionFile);
-        BufferedWriter bw = null;
-        try {
-            bw = new BufferedWriter(new OutputStreamWriter(fos, UTF_8));
+        try (FileOutputStream fos = new FileOutputStream(versionFile);
+             BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(fos, UTF_8))) {
```
Comment (sijie) [inline bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/Cookie.java]: put this in a try block as well

**Comment by sijie** [comment]
run bookkeeper-server bookie tests

**Comment by sijie** [comment]
run bookkeeper-server replication tests

File: bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/Cookie.java (hunk 2 above)
Comment (liketic) [reply bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/Cookie.java]: Thanks @sigee , are you suggest we add a try block for each closable resource here? I think that's won't be necessary. As this [doc](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) said: ```Any resource declared within a try block opening will be closed.```, so one try-block is enough. WDYT?

Comment (sijie) [reply bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/Cookie.java]: I misread the code. you are right. +1 #shipit

**Pull request merged**